core/src/sumeragi/network_topology.rs

```diff
@@ -21,16 +21,16 @@ pub fn sort_peers_by_hash(
 }
 
 /// Sorts peers based on the `hash` and `counter` combined as a seed.
-pub fn sort_peers_by_hash_and_counter(
+fn sort_peers_by_hash_and_counter(
     mut peers: Vec<PeerId>,
     hash: &HashOf<VersionedCommittedBlock>,
     counter: u64,
 ) -> Vec<PeerId> {
     peers.sort_by(|p1, p2| p1.address.cmp(&p2.address));
     let mut bytes: Vec<u8> = counter.to_le_bytes().to_vec();
-    bytes.append(hash.as_ref().to_vec().as_mut());
-    let Hash(bytes) = Hash::new(&bytes);
-    let mut rng = StdRng::from_seed(bytes);
+    bytes.extend(hash.as_ref());
+    let bytes: Hash = HashOf::new(&bytes).into();
+    let mut rng = StdRng::from_seed(bytes.into());
```

Comment:
I made a change here because I don't think it will affect the output. The change is that `bytes` are now encoded before being hashed. This affects the resulting hash because slices are encoded with length prefix